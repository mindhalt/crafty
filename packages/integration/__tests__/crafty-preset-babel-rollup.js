/* global describe, it, expect */

const path = require("path");
const rmfr = require("rmfr");
const testUtils = require("../utils");

// Add a high timeout because of https://github.com/facebook/jest/issues/8942
// Tests would be unreliable if they timeout >_<
jest.setTimeout(30000);

it("Compiles JavaScript with rollup", async () => {
  const cwd = path.join(
    __dirname,
    "../fixtures/crafty-preset-babel-rollup/compiles"
  );
  await rmfr(path.join(cwd, "dist"));

  const result = await testUtils.run(["run", "default"], cwd);

  expect(result).toMatchSnapshot();

  expect(testUtils.exists(cwd, "dist/js/myBundle.min.js")).toBeTruthy();
  expect(testUtils.exists(cwd, "dist/js/myBundle.min.js.map")).toBeTruthy();

  expect(
    testUtils.readForSnapshot(cwd, "dist/js/myBundle.min.js")
  ).toMatchSnapshot();
});

it("Fails gracefully on broken markup", async () => {
  const cwd = path.join(
    __dirname,
    "../fixtures/crafty-preset-babel-rollup/fails"
  );
  await rmfr(path.join(cwd, "dist"));

  const result = await testUtils.run(["run", "default"], cwd);

  expect(result).toMatchSnapshot();

  expect(testUtils.exists(cwd, "dist/js/myBundle.min.js")).toBeFalsy();
  expect(testUtils.exists(cwd, "dist/js/myBundle.min.js.map")).toBeFalsy();
});

it("Lints JavaScript with rollup", async () => {
  const cwd = path.join(
    __dirname,
    "../fixtures/crafty-preset-babel-rollup/lints"
  );
  await rmfr(path.join(cwd, "dist"));

  const result = await testUtils.run(["run", "default"], cwd);

  expect(result).toMatchSnapshot();

  // Files aren't generated on failed lint
  expect(testUtils.exists(cwd, "dist/js/myBundle.min.js")).toBeFalsy();
  expect(testUtils.exists(cwd, "dist/js/myBundle.min.js.map")).toBeFalsy();
});
