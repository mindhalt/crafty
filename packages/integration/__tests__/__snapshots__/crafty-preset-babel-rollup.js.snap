// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Compiles JavaScript with rollup 1`] = `
Object {
  "status": 0,
  "stdall": "
[__:__:__] Starting Crafty __version__...
[__:__:__] Files will be stored in:
            js: __PATH__/packages/integration/fixtures/crafty-preset-babel-rollup/compiles/dist/js
[__:__:__] Starting 'default' ...
[__:__:__] Starting 'js' ...
[__:__:__] Starting 'js_myBundle' ...
[__:__:__] Finished 'js_myBundle' after ____ ms
[__:__:__] Finished 'js' after ____ ms
[__:__:__] Finished 'default' after ____ ms
",
}
`;

exports[`Compiles JavaScript with rollup 2`] = `
"function e(e,t){if(!(e instanceof t))throw new TypeError(\\"Cannot call a class as a function\\")}function t(e,t){for(var n=0;n<t.length;n++){var o=t[n];o.enumerable=o.enumerable||!1,o.configurable=!0,\\"value\\"in o&&(o.writable=!0),Object.defineProperty(e,o.key,o)}}function n(e,n,o){return n&&t(e.prototype,n),o&&t(e,o),e}function o(e){return(o=Object.setPrototypeOf?Object.getPrototypeOf:function(e){return e.__proto__||Object.getPrototypeOf(e)})(e)}function r(e,t){return(r=Object.setPrototypeOf||function(e,t){return e.__proto__=t,e})(e,t)}function i(e,t){return!t||\\"object\\"!=typeof t&&\\"function\\"!=typeof t?function(e){if(void 0===e)throw new ReferenceError(\\"this hasn't been initialised - super() hasn't been called\\");return e}(e):t}var c=function(t){function c(){var t,n;e(this,c);for(var r=arguments.length,u=new Array(r),a=0;a<r;a++)u[a]=arguments[a];return(n=i(this,(t=o(c)).call.apply(t,[this].concat(u)))).handleClick=function(){n.setState({hey:!0})},n}return function(e,t){if(\\"function\\"!=typeof t&&null!==t)throw new TypeError(\\"Super expression must either be null or a function\\");e.prototype=Object.create(t&&t.prototype,{constructor:{value:e,writable:!0,configurable:!0}}),t&&r(e,t)}(c,React.Component),n(c,[{key:\\"render\\",value:function(){return React.createElement(\\"div\\",{onClick:this.handleClick},\\"Hey \\",this.state.hey)}}]),c}(),u=function(){function t(){e(this,t),console.log(\\"init class\\")}return n(t,[{key:\\"method\\",value:function(){console.log(c)}}]),t}();export default u;
//# sourceMappingURL=myBundle.min.js.map
"
`;

exports[`Fails gracefully on broken markup 1`] = `
Object {
  "status": 1,
  "stdall": "
[__:__:__] Starting Crafty __version__...
[__:__:__] Files will be stored in:
            js: __PATH__/packages/integration/fixtures/crafty-preset-babel-rollup/fails/dist/js
[__:__:__] Starting 'default' ...
[__:__:__] Starting 'js' ...
[__:__:__] Starting 'js_myBundle' ...

__PATH__/packages/integration/fixtures/crafty-preset-babel-rollup/fails/js/script.js
  15:1  error  Parsing error: Unexpected token

  13 |
  14 | }
> 15 |
     | ^

✖ 1 problem (1 error, 0 warnings)

[js_myBundle] (!) Errors were found
js/script.js
[__:__:__] 'js_myBundle' errored after ____ ms
[__:__:__] Error: Errors were found
    ...stacktrace...
[__:__:__] 'js' errored after ____ ms
[__:__:__] 'default' errored after ____ ms
",
}
`;

exports[`Lints JavaScript with rollup 1`] = `
Object {
  "status": 1,
  "stdall": "
[__:__:__] Starting Crafty __version__...
[__:__:__] Files will be stored in:
            js: __PATH__/packages/integration/fixtures/crafty-preset-babel-rollup/lints/dist/js
[__:__:__] Starting 'default' ...
[__:__:__] Starting 'js' ...
[__:__:__] Starting 'js_myBundle' ...

__PATH__/packages/integration/fixtures/crafty-preset-babel-rollup/lints/js/script.js
   3:23  error    Delete \`·\`                                       @swissquote/swissquote/prettier/prettier
   5:10  error    'test' is defined but never used                 no-unused-vars
   5:14  error    Replace \`·(one,\` with \`(one,·\`                   @swissquote/swissquote/prettier/prettier
   6:1   error    Delete \`·\`                                       @swissquote/swissquote/prettier/prettier
  11:5   warning  Unexpected console statement                     no-console
  14:9   error    Delete \`·\`                                       @swissquote/swissquote/prettier/prettier
  15:5   warning  Unexpected console statement                     no-console
  15:17  error    Replace \`··\\"some·method\\"·\` with \`\\"some·method\\"\`  @swissquote/swissquote/prettier/prettier
  18:11  error    Replace \`·a,·b)⏎·\` with \`a,·b)\`                  @swissquote/swissquote/prettier/prettier
  20:14  error    Expected '===' and instead saw '=='              eqeqeq

✖ 10 problems (8 errors, 2 warnings)
  6 errors and 0 warnings potentially fixable with the \`--fix\` option.

[js_myBundle] (!) Errors were found
js/script.js
[__:__:__] 'js_myBundle' errored after ____ ms
[__:__:__] Error: Errors were found
    ...stacktrace...
[__:__:__] 'js' errored after ____ ms
[__:__:__] 'default' errored after ____ ms
",
}
`;
